src/runtime/components/CldImage.vue

```diff
@@ -0,0 +1,23 @@
+<script setup lang="ts">
+import type { AssetOptions } from '@cloudinary-util/url-loader'
+
+interface CldImageProps {
+  loading?: "eager" | "lazy";
+  fetchPriority?: "high" | "low" | "auto";
+}
+
+const props = defineProps<AssetOptions & CldImageProps>()
+
+const { url } = useCldImageUrl({ options: props })
+</script>
+
+<template>
+  <img
```

Comment:
does Nuxt Image automatically allow you to just use the <img tag? i guess i expected like <Image or something. this is gonna wrap Nuxt Image right?

Reply:
are we missing any functionality by doing that?

i would have expected us to be able to take advantage of Nuxt Image features like responsive images

by not using it we lose out on a lot of features and end up trying to re-invent the wheel

if you dont want to use Nuxt IMage for some reason, i might suggest then we instead wrap this component:

https://unpic.pics/img/vue/

[this is what we're doing with Svelte](https://github.com/cloudinary-community/svelte-cloudinary/blob/main/svelte-cloudinary/src/lib/components/CldImage.svelte#L87), its a very easy way to get a very powerful component

the reason i would prefer to use Nuxt Image is it would provide less overhead for someone wanting to upgrade form Nuxt Image to Nuxt Cloudinary. change how you import the component and start adding new props. its an easy win

This is what i did with Next Cloudinary and while this doesnt need to exactly follow what I did, by being able to leverage the Next.js Image component, I don't haev to concern myself with building a great image loading experience, i just have to manage creating the URL using Cloudinary tech. its a win-win

and as one last bonus, it would have more of a blessing of doing it the "Nuxt Way" if we're wrapping THE Nuxt Image. its what people expect when building with Nuxt

Reply:
Ok, I will use the NuxtImage then.